### ISSUE TITLE
powershell completion fails with partial word because it includes the incomplete word with completed args

### ISSUE THREAD

**Issue by patricksurry** [fastapi/typer#359]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import typer

valid_names = ["Camila", "Carlos", "Sebastian"]


def complete_name(incomplete: str):
    completion = []
    for name in valid_names:
        if name.startswith(incomplete):
            completion.append(name)
    return completion


def main(
    name: str = typer.Option(
        "World", help="The name to say hi to.", autocompletion=complete_name
    )
):
    typer.echo(f"Hello {name}")


if __name__ == "__main__":
    typer.run(main)
```

### Description

The intro example from the documentation fails in powershell if you start an incomplete name.  For example `typer ./main.py run --name Ca[TAB][TAB]` fails to suggest completions, tho `typer ./main.py run --name [TAB][TAB]` does.  

The [powershell completion](https://github.com/tiangolo/typer/blob/b97298181bd2a932be70ba417b9a712294fa327c/typer/_completion_click8.py#L37) set `args` to the full list of arguments including the incomplete word, e.g. `args = ['--name', 'Ca'], incomplete='Ca'` unlike the [bash completion](https://github.com/tiangolo/typer/blob/b97298181bd2a932be70ba417b9a712294fa327c/typer/_completion_click8.py#L37) which drops the incomplete word, e.g. `args = ['--name'], incomplete='Ca'`.

It looks straightforward to use the [`$cursorPosition`](https://github.com/tiangolo/typer/blob/b97298181bd2a932be70ba417b9a712294fa327c/typer/_completion_shared.py#L56)
 variable to figure out where to truncate the string and take all but the last arg up to that point.  I'll experiment and make a PR.

### Operating System

Windows

### Operating System Details

Windows 11, powershell 7

### Typer Version

0.4.0

### Python Version

3.7.12

**Comment by patricksurry** [comment]
Fixed by https://github.com/tiangolo/typer/pull/360

**Comment by tiangolo** [comment]
Thanks for the report @patricksurry! And thanks for the PR https://github.com/fastapi/typer/pull/360

This fix will be available in Typer `0.12.5`, released in the next few hours. :tada:
